### PR TITLE
Hardening Day One

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by julienlancelot** [comment]
some WS has been forgotten in https://github.com/SonarSource/sonarqube/pull/1582/commits/e5d4c19ab9930424b3f82ad9734b04e9486a0367 => 
- api/permissions/add_group_to_template
- api/permissions/add_project_creator_to_template
- api/permissions/add_user_to_template
- api/permissions/create_template
- api/permissions/delete_template
- api/permissions/remove_group_from_template
- api/permissions/remove_project_creator_from_template
- api/permissions/remove_user_from_template 
- ...

**Comment by julienlancelot** [comment]
This WS has been forgotten in https://github.com/SonarSource/sonarqube/pull/1582/commits/5bb73df469735123361e008bceeb8655b4c91de8 : 
- api/user_groups/users

**Comment by julienlancelot** [comment]
WS has been forgotten in https://github.com/SonarSource/sonarqube/pull/1582/commits/ac074ced86a9ea11b2af0beaf049e9c47e7c567e : 
- api/components/show

**Comment by julienlancelot** [comment]
Ws has been forgotten in https://github.com/SonarSource/sonarqube/pull/1582/commits/f2ab231ef7f0dacd4ffa5714f2f1c3e36d769c2e : 
- api/measures/search_history

**Comment by teryk** [comment]
feedbacks applied

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchHistoryAction.java [hunk 1]
```diff
@@ -73,12 +73,28 @@ public SearchHistoryAction(DbClient dbClient, ComponentFinder componentFinder, U
...
   @Override
   public void define(WebService.NewController context) {
     WebService.NewAction action = context.createAction(ACTION_SEARCH_HISTORY)
       .setDescription("Search measures history of a component.<br>" +
         "Measures are ordered chronologically.<br>" +
-        "Pagination applies to the number of measures for each metric.")
+        "Pagination applies to the number of measures for each metric.<br>" +
+        "Requires one of the following permissions:" +
+        "<ul>" +
+        "<li>'Administer System'</li>" +
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchHistoryAction.java]: It's wrong, only Browser permission is required

**Comment by julienlancelot** [comment]
Last feedback to apply and LTM !

**Pull request merged**